b2luigi/batch/processes/gbasf2.py

```diff
@@ -410,7 +410,11 @@ def _build_gbasf2_submit_command(self):
 
         gbasf2_input_dataset = get_setting("gbasf2_input_dataset", default=False, task=self.task)
         if gbasf2_input_dataset is not False:
-            gbasf2_command_str += f" -i {gbasf2_input_dataset} "
+            if ".txt" not in gbasf2_input_dataset:
```

Comment:
I would try to be more explicit here with the [string.endswith](https://www.tutorialspoint.com/python/string_endswith.htm) method:
```suggestion
            if not gbasf2_input_dataset.endswith(".txt"):
```
- [ ] replace `in` with `.endswidth`

Reply:
Hi @meliache

Yeah, I've actually noticed shortly after my pull request, that you have already done some work in #75 on that issue (see my comment on it).

I'm fine with both ways to solve that & can test the branch in #75 if you like, since I am currently anyhow working on a suitable use-case: https://github.com/ArturAkh/b2luigi/tree/fei_with_gbasf2

In that case, I have a working environment with all the needed software.

Please just decide which solution should be taken to the main branch and I'll either test the #75 one or extend the needed "help" message adaption & the suggested code change.

Reply:
About your solution I just don't like that in theory the `--input_dslist` works also with LFN files that have other endings than `.txt`, e.g. I know some people [use[(https://questions.belle2.org/question/8334/is-it-possible-to-process-one-experiment-on-the-grid-by-one-command/) `.lfn`. So someone might provide the path to their local text file with another file ending and wonder why they get an error that the dataset doesn't exist.

About my solution I I don't like that I add yet another setting and the complexity of having an `xor`-like settings requirement. So I need to sleep a night over this and maybe ask someone.